Add 'count' option to the 'RANDOMKEY' command

hwware opened the pull request:
Regarding to https://github.com/redis/redis/issues/11915

Requirement:

Now RANDOMKEY command return a random key from the currently selected database OR nil when the database is empty
We want to enhancement this command for more features:

Add count option to specified number of keys
Add duplicate option to allow return duplicated or non-duplicated random key
Add pattern option to allow client to specified the key pattern.
The command format could be like this:

RANDOMKEY [COUNT count] [Duplicated ] [Pattern XXXX]
Default: count = 1; non-duplicated; and no patten which means must match exactly keyname

hwware commented:
Note: Took the reference of ZRANDMEMBER and SRANDMEMBER specifications for duplicate option.

ncghost1 commented on src/db.c:
```diff
@@ -751,17 +751,140 @@ void selectCommand(client *c) {
-
+/************************************************************
+RANDOMKEY [COUNT <count> [DUPLICATED] [PATTERN <pattern>]] (OR)
+RANDOMKEY [COUNT <count> [DUPLICATED]
```
```suggestion
RANDOMKEY [COUNT <count> [DUPLICATED]]
```

hwware replied:
Added changes.

Thanks!

ncghost1 reviewed (commented):
Thank you, you have added all the features I want. : )

oranagra reviewed (commented):
this could be a candidate for Redis 8.0, so i won't do a detailed review yet.
from a quick look i have a few observations:
1. this mixes the complications of SRANDMEMBER (Unique, and CLOSE_ASAP), with the complications of SCAN (pattern matching, expiration, and maxtries).
2. i'm afraid of the execution complexity, specifically in edge cases (it should be ok for it to return empty handed even if there are keys that can be returned), and it should probably do bigger effort to avoid long loops, like the way `server.masterhost` is used in dbRandomKey) 
3. i wonder if the interface should be a bit more similar to SRANDMEMBER (with the negative count), i guess not.
4. i'm not sure i'm happy with the current code is how arguments are handled. first the fact that we have two forms for the command, and not handing each argument on it's own regardless of order. secondly it seems that the argument parsing code is mixed with the execution part (which should no longer look at the argc and argv IMHO)
5. also, the fact that `dbRandomKey` is called even before parsing argument.
6. reply schema in the json file is missing, but that also makes me wonder if it should be a new separate command, or an extension to the existing command.

hwware commented:
Hi @oranagra 

Thank you for your review! Based on your comments, updated the new patch which has below changes.
-Changed the logic of parsing the arguments.
-Removed the part of code where argument parsing along with the execution.
-Added the flags for argument options and used same flags for the execution part.
-Also added checks as mentioned to avoid the long loops.
-dbRandomKey was used before argument parsing to check empty db initially, my mistake I forgot to update with direct db check. Now its updated accordingly.
-Added logic for negative count values to be consistent with other commands.
-Also updated the json file accordingly.

hwware commented:
@oranagra Could you please review again when you get sometime.

hwware commented:
Hi @oranagra, could you please take a look this PR as a candidate for Redis 8.0 when you have time? Thank you!
